bc_obps/reporting/api/facility_report.py

```diff
@@ -0,0 +1,66 @@
+from typing import Literal, Tuple, Optional
+from uuid import UUID
+from django.http import HttpRequest
+from registration.decorators import handle_http_errors
+from reporting.constants import EMISSIONS_REPORT_TAGS
+from reporting.schema.generic import Message
+from service.facility_report_service import FacilityReportService
+from service.error_service.custom_codes_4xx import custom_codes_4xx
+from .router import router
+from reporting.schema.facility_report import FacilityReportOut, FacilityReportIn
+
+
+@router.get(
+    "/report-version/{version_id}/facility-report/{facility_id}",
+    response={200: FacilityReportOut, 404: Message, 400: Message, 500: Message},
+    tags=EMISSIONS_REPORT_TAGS,
+    description="""Takes `version_id` (primary key of the ReportVersion model) and `facility_id` to return a single matching `facility_report` object.
+    Includes the associated activity IDs if found; otherwise, returns an error message if not found or in case of other issues.""",
+)
+@handle_http_errors()
+def get_facility_report_form_data(
+    request: HttpRequest, version_id: int, facility_id: UUID
+) -> Tuple[Literal[200], Optional[FacilityReportOut]]:
+    facility_report = FacilityReportService.get_facility_report_by_version_and_id(version_id, facility_id)
+    activity_ids = FacilityReportService.get_activity_ids_for_facility(version_id, facility_id) or []
```

Comment:
The service already returns an empty list if none are found
```suggestion
    activity_ids = FacilityReportService.get_activity_ids_for_facility(version_id, facility_id)
```